system/include/emscripten/bind.h

```diff
@@ -1875,6 +1875,21 @@ class class_ {
     }
 };
 
+#if __cplusplus >= 201703L
+template<typename T>
+void register_optional() {
+    // Optional types are automatically registered for some internal types so
+    // only run the register method once so we don't conflict with a user's
+    // bindings if they also register the optional type.
+    static bool hasRun;
```

Comment:
Should this be `thread_local` so that optionals can be registered in new Workers?